core/dbt/graph/cli.py

```diff
@@ -66,7 +67,7 @@ def parse_union_from_default(
 def parse_difference(
     include: Optional[List[str]], exclude: Optional[List[str]]
 ) -> SelectionDifference:
-    included = parse_union_from_default(include, DEFAULT_INCLUDES)
+    included = parse_union_from_default(include, DEFAULT_INCLUDES, greedy=flags.GREEDY)
```

Comment:
This captures all standard selection (`--select`, `--exclude`) perfectly.

To make the `--greedy` flag work with a yaml `--selector` too, we'd need to add a similar bit of logic that passes `flags.GREEDY` into each method called by [`parse_from_definition`](https://github.com/dbt-labs/dbt/blob/51a76d0d63efd8a393ace482c6c93afcf75f446c/core/dbt/graph/cli.py#L245). That's fairly straightforward for two of the conditions—`SelectionCriteria.from_single_spec(definition, greedy=flags.GREEDY)` and `SelectionCriteria.selection_criteria_from_dict(definition, dct, greedy=flags.GREEDY)` (called by `parse_dict_definition`) already accept `greedy` as an argument—but we'd need to add `greedy` arg support for `parse_union_definition` and `parse_intersection_definition` as well.

So: I'm thinking more about yaml selectors, and how they intentionally _don't_ (or _shouldn't_) intersect with additional CLI-provided selection criteria, and about the fact that they can define `greedy` themselves (!). I'll say more in my overall review comment.